engine/script/src/script.cpp

```diff
@@ -1681,23 +1807,27 @@ namespace dmScript
         if(cbk->m_ContextTableRef != LUA_NOREF)
         {
             lua_rawgeti(L, LUA_REGISTRYINDEX, cbk->m_ContextTableRef);
-            if (lua_type(L, -1) == LUA_TTABLE)
+            if (lua_type(L, -1) == LUA_TTABLE && IsCallbackInstanceValid(cbk))
             {
                 // We do not use dmScript::Unref for refs in the context local table as we don't
                 // want to count those refs the ref debug count shown in the profiler
                 luaL_unref(L, -1, cbk->m_Self);
                 luaL_unref(L, -1, cbk->m_Callback);
-
-                // For the callback (that can actually outlive the script instance)
-                // we want to add to the lua debug count
+            }
+            
+            // For the callback (that can actually outlive the script instance)
+            // we want to add to the lua debug count
+            if (cbk->m_CallbackInfoRef != LUA_NOREF)
+            {
                 dmScript::Unref(L, LUA_REGISTRYINDEX, cbk->m_CallbackInfoRef);
```

Comment:
we call `dmScript::Ref` for  `cbk->m_CallbackInfoRef` separatly from `cbk->m_ContextTableRef` when we create callback. So, even if callback isn't valid, this Ref still exist and we have unref it.